sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala

```diff
@@ -78,7 +79,8 @@ class HiveSessionStateBuilder(session: SparkSession, parentState: Option[Session
         customResolutionRules
 
     override val postHocResolutionRules: Seq[Rule[LogicalPlan]] =
-      new DetermineTableStats(session) +:
+      new DetectAmbiguousSelfJoin(conf) +:
```

Comment:
How about adding a test case for ensuring all these rules are included? So far, we do not have a test case for covering it if the rule is missing to add.

Reply:
I need to know the details of each rule so that I can write a test to make sure these rules are included. Maybe do it in a followup instead of blocking this PR? BTW the new tests added in this PR do guarantee the `DetectAmbiguousSelfJoin` rule is there.

Reply:
Created a ticket https://issues.apache.org/jira/browse/SPARK-28629